controllers/cloudstackcluster_controller_test.go

```diff
@@ -17,70 +17,39 @@ limitations under the License.
 package controllers_test
 
 import (
-	"context"
-	"time"
-
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"sigs.k8s.io/cluster-api-provider-cloudstack/test/dummies"
-	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
-	"sigs.k8s.io/cluster-api/util/patch"
+	infrav1 "sigs.k8s.io/cluster-api-provider-cloudstack/api/v1beta2"
+	"sigs.k8s.io/cluster-api-provider-cloudstack/controllers"
+	dummies "sigs.k8s.io/cluster-api-provider-cloudstack/test/dummies/v1beta2"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-const (
-	timeout = time.Second * 30
-)
-
 var _ = Describe("CloudStackClusterReconciler", func() {
-	ctx := context.Background()
-
-	BeforeEach(func() {
-		dummies.SetDummyVars()
+	Context("With k8s like test environment.", func() {
+		BeforeEach(func() {
+			SetupTestEnvironment()                                                    // Must happen before setting up managers/reconcilers.
+			Ω(ClusterReconciler.SetupWithManager(k8sManager)).Should(Succeed())       // Register CloudStack ClusterReconciler.
+			Ω(FailureDomainReconciler.SetupWithManager(k8sManager)).Should(Succeed()) // Register CloudStack ClusterReconciler.
+		})
+
+		It("Should create a CloudStackFailureDomain.", func() {
+			tempfd := &infrav1.CloudStackFailureDomain{}
+			Eventually(func() bool {
+				key := client.ObjectKeyFromObject(dummies.CSFailureDomain1)
+				key.Name = key.Name + "-" + dummies.CSCluster.Name
+				if err := k8sClient.Get(ctx, key, tempfd); err != nil {
+					return true
+				}
+				return false
+			}, timeout).Should(BeTrue())
+		})
 	})
 
-	It("Should create a cluster", func() {
-		By("Fetching a CS Cluster Object")
-		// TODO make the tests work with other CRDs. This stage can't be reached yet.
-		// CS.EXPECT().GetOrCreateCluster(gomock.Any()).MinTimes(1)
```

Comment:
All of the stuff removed from here still exists in some form, but has been moved to the suite setup file.